app/lib/utils/gpDataMapper.js

```diff
@@ -7,7 +7,7 @@ function getFilteredGps(gpData, searchStr) {
   if (gpData.doctors) {
     let filteredSearchStr = searchStr;
     let filteredDocs = [];
-    if (searchStr.match(/^DOCTOR|Doctor|doctor|DR|Dr.|Dr|dr.|dr/)) {
+    if (searchStr.match(new RegExp(/^(doctor|dr.|dr)/, 'i'))) {
```

Comment:
Use the literal form of regex creation as opposed to constructor form when the input isn't runtime specific